File: 876. Middle of the Linked List.py
# Name = Vipan Kumar
# GitHub user name = @VipanKumar01



class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:

        tort = hare = head
        
        while hare and hare.next:
            tort = tort.next
            hare = hare.next.next
            
        return tort
      
# -- Happy Code --

File: 876. Middle of the Linked List.c
// I am Already solve that in Python But I study about DSA of C it is Same As Python So I am Posting of that Code 

struct ListNode* middleNode(struct ListNode* head){
    struct ListNode* hare = head;   
    // 2X Faster 
    struct ListNode* tort = head;  
    // 1X faster
    
    while (hare != NULL)
	{
		if (hare->next != NULL)
		{
			hare = hare->next->next;
			tort = tort->next;
		}
		else hare = hare->next;
	}

	return tort;
}

// --HappyCode--
